lib/logstash/codecs/fluent.rb

```diff
@@ -116,7 +117,7 @@ def decode_event(data, &block)
         raise(LogStash::Error, "PackedForward with compression is not supported")
       end
 
-      entries_decoder = @factory.unpacker
+      entries_decoder = (@packforward_decoder ||= @factory.unpacker).tap(&:reset)
```

Comment:
As I understand `decode`, when used by a input like TCP (which is multithreaded), is used by multiple threads, so I have feeling that `@packforward_decoder.reset` could be called and used by multiple threads.
WDYT?

Reply:
TCP clones the codec, and gives a fresh codec to each connection, so this should never be shared.